src/software/amazon/ionschema/internal/util/StringExtensions.kt

```diff
@@ -0,0 +1,12 @@
+package software.amazon.ionschema.internal.util
+
+/**
+ * String extension functions
+ */
+internal fun String.truncate(limit: Int, truncated: CharSequence = "..."): String {
+    if (this.length < limit) {
+        return this
+    }
+    return this.substring(0, limit) + truncated
```

Comment:
this fun follows the pattern of [joinToString](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html), e.g.:
```
charArrayOf('a', 'b', 'c', 'd', 'e').joinToString(limit = 3, truncated = "...", separator = "")
```

produces `abc...`

keeping as-is

Reply:
since this is only being used for [error messages](https://github.com/amzn/ion-schema-kotlin/blob/6913ed40f458cf92f1c5f58a14c8e94d11f2c8ed/src/software/amazon/ionschema/Violations.kt#L87) I don't see a problem leaving as is